### PR TITLE
xds: handle wildcard TypeURI

### PR DESCRIPTION
**Pull request by eduser25**
Handle wildcard xDS proto.

Signed-off-by: edu <eduser25@gmail.com>

- Does this change contain code from or inspired by another project? If so, did you notify the maintainers and provide attribution?
No

### CODE REVIEW

File: pkg/envoy/ads/response.go [hunk 1]
```diff
@@ -65,7 +65,7 @@ func (s *Server) sendResponse(typeURIsToSend mapset.Set,
 
 		// Handle request when is not provided, and the SDS case
 		var finalReq *xds_discovery.DiscoveryRequest
-		if request == nil {
+		if request == nil || request.TypeUrl == string(envoy.TypeWildcard) {
```
Comment (shashankram) [inline pkg/envoy/ads/response.go:68]: Would change `string(envy.TypeWildcard)` to `envoy.TypeWildcard.String()` and add `String()` method receiver for the type.

File: pkg/envoy/ads/stream.go [hunk 2]
```diff
@@ -172,7 +172,20 @@ func (s *Server) StreamAggregatedResources(server xds_discovery.AggregatedDiscov
...
-			err := s.sendResponse(mapset.NewSetWith(typeURL), proxy, &server, &discoveryRequest, s.cfg)
+			var xdsUpdatePaths mapset.Set
+			switch typeURL {
+			case envoy.TypeWildcard:
+				xdsUpdatePaths = mapset.NewSetWith(
+					envoy.TypeCDS,
+					envoy.TypeEDS,
+					envoy.TypeLDS,
+					envoy.TypeRDS,
+					envoy.TypeSDS)
```
Comment (shashankram) [inline pkg/envoy/ads/stream.go:183]: Does order matter here? If it does, does mapset gaurantee ordering (I guess not)?

**Review by shashankram — APPROVED** [review]

File: pkg/envoy/ads/response.go (hunk 1 above)
Comment (eduser25) [reply pkg/envoy/ads/response.go:68]: can do. Not that I remember, I think that was the VC autocompletion

File: pkg/envoy/ads/stream.go (hunk 2 above)
Comment (eduser25) [reply pkg/envoy/ads/stream.go:183]: (1) i think order is not guaranteed (i think it uses internal map?) and (2) order here doesn't matter

Order is however guaranteed when calling `sendResponse` over [here](https://github.com/openservicemesh/osm/pull/2938/files#diff-7a1f450fbabb0088828083ed90ad3ccc49ed4edd9ae6f58ad5bdcdd158b2dda8R62)

**Comment by eduser25** [comment]
scale test was run ffor this commit https://github.com/eduser25/osm/actions/runs/668859481

**Review by snehachhabria — APPROVED** [review]

**Pull request merged**